### PR TITLE
fix: 'TypeError: o[0].rules[0] is undefined' when trying to execute function

### LINKED ISSUE #1360: 🐛 Bug Report: 'TypeError: o[0].rules[0] is undefined' when trying to execute function
### 👟 Reproduction steps

Go to a function and press Execute in the console.

### 👍 Expected behavior

The execution window should show.

### 👎 Actual Behavior

The execution window does not show and the page content disappears.
Logs:
```
Uncaught (in promise) TypeError: o[0].rules[0] is undefined
    Immutable 43
    <anonymous> https://website.com/console/project-0/functions/function:187
    promise callback* https://website.com/console/project-0/functions/function:186
    promise callback* https://website.com/console/project-0/functions/function:180
```
Error in `DeploymentDomains.XXX.js`
bild

### 🎲 Appwrite version

Different version (specify in environment)

### 💻 Operating system

Linux

### 🧱 Your Environment

Appwrite 1.6.0
Appwrite console 5.0.12
Self-hosted
No function domains

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by askadityapandey**
## What does this PR do?

Fixes an issue where the execution window does not show and the page content disappears when executing a function in the console.

## Test Plan
1. Go to a function.
2. Press Execute in the console.

## Related PRs and Issues
fixes #1360
### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes.

### CODE REVIEW

**Comment by stnguyen90** [comment]
@askadityapandey, please make sure you comment on the related issue and ask to work on it so that multiple people don't work on the same issue, leading to wasted work.

**Comment by askadityapandey** [comment]
> @askadityapandey, please make sure you comment on the related issue and ask to work on it so that multiple people don't work on the same issue, leading to wasted work.

Okay, but it'll already show the linked PR🙏😭

File: src/routes/(console)/project-[project]/functions/function-[function]/deploymentDomains.svelte [hunk 1]
```diff
@@ -8,31 +8,38 @@
...
+                                icon="external-link">
+                                {rule.domain}
+                            </DropListLink>
+                        {/if}
+                    {/each}
+                </svelte:fragment>
+            </DropList>
+        {/if}
+    {:else}
+        <p>No rules available</p>
```
Comment (stnguyen90) [inline src/routes/(console)/project-[project]/functions/function-[function]/deploymentDomains.svelte:43]: Let's change this to domains as rules wouldn't make sense for most people.

```suggestion
        <p>No domains available</p>
```

**Review by stnguyen90 — COMMENTED** [review]
> Okay, but it'll already show the linked PR

I'm unable to assign the issue to you, though.

File: src/routes/(console)/project-[project]/functions/function-[function]/deploymentDomains.svelte [hunk 2]
```diff
@@ -8,18 +8,21 @@
...
-        class="u-flex u-gap-4 u-cross-center">
-        <Trim alternativeTrim>
-            <span class="link">
-                {domain.rules[0].domain}
-            </span>
-        </Trim>
-        <span class="icon-external-link" aria-hidden="true" />
-    </a>
-    {#if domain.rules.length > 1}
+    {#if domain && domain.rules && domain.rules.length > 0}
```
Comment (TorstenDittmann) [inline src/routes/(console)/project-[project]/functions/function-[function]/deploymentDomains.svelte:11]: can we make this a bit cleaner by having a variable in the script? 👀 

something like 
```js
$: length: domain?.rules?.length
```

then we can have more simple conditions 👍🏻

**Review by TorstenDittmann — CHANGES_REQUESTED** [review]
Thank you for the PR, just a small thing to change :)

**Comment by stnguyen90** [comment]
After discussing internally, we want to go with hiding the section (Domains title and URL) altogether if there is no domain.

Thanks for your efforts, but we're going to close this and handle it ourselves. Next time, please comment on the issue and ask to be assigned before submitting a PR to avoid wasted work.

**Pull request closed without merging**